accepted/future-releases/enhanced-enums/feature-specification.md

```diff
@@ -68,128 +66,175 @@ must redirect to a generative constructor._
+
+The semantics of such an enum declaration, *E*, is defined as introducing a (semantic) *class*, *C*, just like a similar `class` declaration.
+
+* **Name**: The name of the class *C* and its implicit interface is the name of the `enum` declaration.
+
+* **Superclass**: The superclass of *C* is (seemingly) the `Enum` class, with the mixins declared by *E* applied. _(In practice, the superclass may be another system provided class with implements `Enum`.)_
+
+  * If *E* is declared as `enum Name with Mixin1, Mixin2 …` then the superclass of *C* is the mixin application `Enum with Mixin1, Mixin2`.
+
+  It’s a **compile-time error** if such a mixin application introduces any instance variables. _We need to be able to call an implementation specific superclass `const` constructor of `Enum`, and a mixin application of a mixin with a field does not make its forwarding constructor `const`. Currently that’s the only restriction, but if we add further restrictions on mixin applications having `const` forwarding constructors, those should also apply here._
+
+* **Superinterfaces**: The immediate superinterfaces of *C* are the interface of the superclass and the interfaces declared by *E*.
+  * If `E` is declares as `enum Name with Mixin1, Mixin2 implements Type1, Type2 { … }` then the immediate superinterfaces of *C* are the interfaces of `Name with Mixin1, Mixin2`, `Type1` and `Type2`.
+
+- **Declared members**: For each member declaration of the `enum` declaration *E*, the same member is added to the class *C*. This includes constructors (which must be `const` generative or non-`const` factory constructors.)
+
+- **Default constructor**: If no generative constructors were declared, and no unnamed factory constructor was added,
+  a default generative constructor is added:
+
+  ```dart
+  const Name();
+  ```
+  
+  _(This differs from the default constructor of a normal `class` declaration by being constant.)_
+  
+- **Enum values**: For each `<enumEntry>` with name `id` and index *i* in the comma-separated list of enum entries, a constant value is created, and a static constant variable named `id` is created in *C* with that value. All the constant values are associated, in some implementation dependent way, with 
+  
+- their name `id` as a string `"id"`, 
+  - their index *i* as an `int`, and
+  - their `enum` class’s name as a string, `"Name"`,
+  
+  all of which is accessible to the `toString` and `index` member of `Enum`, and to the `EnumName.name` extension getter. The values are computed as follows.
+  
+  - `id` &mapsto; `Name()` (no arguments, equivalent to empty argument list)
+  - `id(args)` &mapsto; `Name(args)`
+  - `id<types>(args)` &mapsto; `Name<types>(args)`
+  - `id.named(args)` &mapsto; `Name._$named(args)`
+  - `id<types>.named(args)` &mapsto; `Name<types>._$named(args)`
+  
+  where `args` are considered as occurring in a `const` context, and it’s a **compile-time error** if they are then not compile-time constants.
```

Comment:
Arguments that occur in a constant context without any `const` in a syntactically enclosing term seems to be a novel thing, but I guess it's OK because it _does_ occur in the desugared declaration.

Reply:
ACK.